test/LibraryManager.IntegrationTest/AddCientSideLibrariesFromUITests.cs

```diff
@@ -0,0 +1,120 @@
+
+            if (libManConfig != null)
+            {
+                libManConfig.Open();
+
+                Editor.Selection.SelectAll();
+                Editor.KeyboardCommands.Delete();
+                Editor.Edit.InsertTextInBuffer(_initialLibmanFileContent);
+
+                libManConfig.Save();
+            }
+
+            base.DoHostTestCleanup();
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromProjectRoot_SmokeTest()
+        {
+            SetLibraryAndClickInstall(_projectName, "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/lib/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        [TestMethod]
+        public void InstallClientSideLibraries_FromFolder_SmokeTest()
+        {
+            SetLibraryAndClickInstall("wwwroot", "jquery-validate@1.17.0");
+
+            string pathToLibrary = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "jquery-validate");
+            string[] expectedFiles = new[]
+            {
+                Path.Combine(pathToLibrary, "jquery.validate.js"),
+                Path.Combine(pathToLibrary, "localization", "messages_ar.js"),
+            };
+
+            string manifestContents = @"{
+  ""version"": ""1.0"",
+  ""defaultProvider"": ""cdnjs"",
+  ""libraries"": [
+    {
+      ""library"": ""jquery-validate@1.17.0"",
+      ""destination"": ""wwwroot/jquery-validate/""
+    }
+  ]
+}";
+            Helpers.FileIO.WaitForRestoredFiles(pathToLibrary, expectedFiles, caseInsensitive: true, timeout: 20000);
+            Assert.AreEqual(manifestContents, File.ReadAllText(_pathToLibmanFile));
+        }
+
+        private void SetLibraryAndClickInstall(string nodeName, string library)
+        {
+            SolutionExplorerItemTestExtension solutionExplorerItemTestExtension = SolutionExplorer.FindItemRecursive(nodeName);
+            solutionExplorerItemTestExtension.Select();
+
+            InstallDialogTestService installDialogTestService = VisualStudio.Get<InstallDialogTestService>();
+            InstallDialogTestExtension installDialogTestExtenstion = installDialogTestService.OpenDialog();
+
+            installDialogTestExtenstion.SetLibrary(library);
+            installDialogTestExtenstion.WaitForFileSelections();
```

Comment:
>WaitForFileSelections [](start = 40, length = 21)

Is this going to be a frequent pattern?

SetLibrary()
WaitForFileSelection()  // will this always be required?
ClickInstall()

I'm just curious what we get out of the wait, or if it's something that always needs to happen (in which case, can we push it into SetLibrary so that callers don't need to know about it)?

Reply:
I left it outside initially thinking it would change when we add more tests. For now I think it makes sense to move inside SetLibrary

---
In reply to: [206380380](https://github.com/aspnet/LibraryManager/pull/332#discussion_r206380380) [](ancestors = 206380380)